guidance/src/guidance/guidance_worker.cpp

```diff
@@ -18,62 +18,65 @@
 
 namespace guidance
 {
-    GuidanceWorker::GuidanceWorker() 
+    GuidanceWorker::GuidanceWorker() : called_robotic_engage_in_active_(false)
     {
         nh_ = ros::CARMANodeHandle{};
         pnh_ = ros::CARMANodeHandle{"~"};
     }
 
     void GuidanceWorker::system_alert_cb(const cav_msgs::SystemAlertConstPtr& msg)
     {
-        gsm->onSystemAlert(msg);
+        gsm_.onSystemAlert(msg);
     }
 
     void GuidanceWorker::robot_status_cb(const cav_msgs::RobotEnabledConstPtr& msg)
     {
-        gsm->onRoboticStatus(msg);
+        gsm_.onRoboticStatus(msg);
     }
 
     bool GuidanceWorker::guidance_acivation_cb(cav_srvs::SetGuidanceActiveRequest& req, cav_srvs::SetGuidanceActiveResponse& res)
     {
         // Translate message type from GuidanceActiveRequest to SetEnableRobotic
         ROS_INFO_STREAM("Request for guidance activation recv'd with status " << req.guidance_active);
-        gsm->onSetGuidanceActive(req.guidance_active);
-        cav_srvs::SetEnableRobotic srv;
-        if (gsm->getCurrentState() == GuidanceStateMachine::ENGAGED) {
-            srv.request.set = cav_srvs::SetEnableRobotic::Request::ENABLE;
-            res.guidance_status = true;
-        } else {
+        if(!req.guidance_active)
+        {
+            cav_srvs::SetEnableRobotic srv;
             srv.request.set = cav_srvs::SetEnableRobotic::Request::DISABLE;
-            res.guidance_status = false;
+            enable_client_.call(srv);
         }
-        enable_client_.call(srv);
+        gsm_.onSetGuidanceActive(req.guidance_active);
+        res.guidance_status = req.guidance_active;
```

Comment:
Shouldn't we set the guidance_status based on the result from onSetGuidanceActive

Reply:
Good catch!